dataline-scheduler/src/test/java/io/dataline/scheduler/JobSubmitterTest.java

```diff
@@ -25,65 +25,120 @@
 package io.dataline.scheduler;
 
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
 import static org.mockito.Mockito.when;
 
+import com.google.common.util.concurrent.MoreExecutors;
 import io.dataline.scheduler.persistence.SchedulerPersistence;
+import io.dataline.workers.JobStatus;
+import io.dataline.workers.OutputAndStatus;
 import java.io.IOException;
 import java.util.Optional;
-import java.util.concurrent.ExecutorService;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.InOrder;
 
 public class JobSubmitterTest {
 
-  private static final Job JOB = new Job(
-      1L,
-      "",
-      JobStatus.PENDING,
-      null,
-      null,
-      null,
-      null,
-      1L,
-      null,
-      1L);
-
-  private ExecutorService executorService;
+  public static final OutputAndStatus<Integer> SUCCESS_OUTPUT = new OutputAndStatus<>(JobStatus.SUCCESSFUL, 1);
+  public static final OutputAndStatus<Integer> FAILED_OUTPUT = new OutputAndStatus<>(JobStatus.FAILED);
+
   private SchedulerPersistence persistence;
+  private WorkerRunFactory workerRunFactory;
+  private WorkerRun workerRun;
+
   private JobSubmitter jobSubmitter;
 
   @BeforeEach
-  public void setup() {
-    executorService = mock(ExecutorService.class);
+  public void setup() throws IOException {
+    Job job = mock(Job.class);
+    when(job.getId()).thenReturn(1L);
     persistence = mock(SchedulerPersistence.class);
+    when(persistence.getOldestPendingJob()).thenReturn(Optional.of(job));
+
+    workerRun = mock(WorkerRun.class);
+    workerRunFactory = mock(WorkerRunFactory.class);
+    when(workerRunFactory.create(job)).thenReturn(workerRun);
 
     jobSubmitter = new JobSubmitter(
-        executorService,
-        null,
+        MoreExecutors.newDirectExecutorService(),
         persistence,
-        null,
-        null);
+        workerRunFactory);
+  }
+
+  @Test
+  public void testPersistenceNoJob() throws Exception {
+    doReturn(Optional.empty()).when(persistence).getOldestPendingJob();
```

Comment:
is there a reason you prefer this syntax to `when(persistence.getOldestPendingJob()).thenReturn(Optional.empty());`?

Reply:
not really.